core/src/avm2/object/class_object.rs

```diff
@@ -368,6 +352,45 @@ impl<'gc> ClassObject<'gc> {
         Ok(())
     }
 
+    // Looks up an interface by name, without using `ScopeChain.resolve`
+    // This lets us look up an interface before its `ClassObject` has been constructed,
+    // which is needed to resolve interfaces when constructing a (different) `ClassObject`.
+    fn early_resolve_interface(
+        &self,
+        scope: ScopeChain<'gc>,
```

Comment:
Does this method need to be taking `scope` as an argument? It might be simpler for the method to get the class scope itself

Reply:
I've changed this to directly take in a `Domain`